src/main/java/com/uddernetworks/holysheet/SheetManager.java

```diff
@@ -41,17 +42,62 @@ public SheetManager(Drive drive, Sheets sheets) {
         this.sheetIO = new SheetIO(this, drive, sheets);
     }
 
+    /**
+     * Return a {@link File} from drive with the id provided.
+     *
+     * @param id The id of the file.
+     * @return {@link File} if found, otherwise {@code null}.
+     * @throws IOException An exception from the drive request.
+     */
     public File getFile(String id) throws IOException {
-        return drive.files().get(id).execute();
+        try {
+            return drive.files().get(id).execute();
+        } catch (GoogleJsonResponseException exception) {
+            if (exception.getStatusCode() == 404) {
+                return null;
+            } else throw exception;
+        }
     }
 
+    /**
+     * Return a {@link File} from drive with the id provided; and the fields
+     * to request from a partial response from Drive.
+     *
+     * @param id The id of the file.
+     * @return {@link File} if found, otherwise {@code null}.
+     * @throws IOException An exception from the drive request.
+     */
     public File getFile(String id, String fields) throws IOException {
         return drive.files().get(id).setFields(fields).execute();
     }
 
+    /**
+     * Return the id of a google drive file matching the given name, not necessarily inside the sheetStore folder.
+     *
+     * @param name The name to match.
+     * @return {@link Optional} wrapping the {@link String} name.
+     */
     public Optional<String> getIdOfName(String name) {
+        return getIdOfName(name, true);
+    }
+
+    /**
+     * Return the id of a google drive file matching the given name.
+     *
+     * @param name         The name to match.
+     * @param inSheetStore Whether this file must be in the sheetStore folder.
+     * @return {@link Optional} wrapping the {@link String} name.
+     */
+    public Optional<String> getIdOfName(String name, boolean inSheetStore) {
```

Comment:
> inSheetStore Whether this file must be in the sheetStore folder.
In what case would this be true? I feel like for consistency purposes that should always be true, no? Other parts of the program could rely on the listing of files from there. Then again they already search for the parent property.

Reply:
That was used in the cloning. The reason cloning wasn't working is because the method only searches in the _sheetStore_ folder, not very useful for cloning outside of the folder. So, I made the method by default only search in that folder, if specified it'd search their entire drive. So by default it's true, which is what it was before implicitly.

Reply:
That makes sense, thank you